GhostNetwork.Messages.Api/Controllers/MessageController.cs

```diff
@@ -0,0 +1,108 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Swashbuckle.AspNetCore.Filters;
+
+namespace GhostNetwork.Messages.Api.Controllers;
+
+[ApiController]
+[Route("[controller]")]
+public class MessageController : ControllerBase
+{
+    private readonly IMessageService _messageService;
+
+    public MessageController(IMessageService messageService)
+    {
+        _messageService = messageService;
+    }
+
+    /// <summary>
+    /// Get messages by chat id
+    /// </summary>
+    /// <param name="chatId">chat id</param>
+    /// <param name="skip">Skip exist messages up to a specified position</param>
+    /// <param name="take">Take exist messages up to a specified position</param>
+    /// <response code="200">Chat messages</response>
+    [HttpGet("{chatId:guid}")]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [SwaggerResponseHeader(StatusCodes.Status200OK, "X-TotalCount", "Number", "Total number of messages")]
+    public async Task<ActionResult<IEnumerable<Message>>> SearchAsync(
+        [FromRoute] Guid chatId,
+        [FromQuery, Range(0, int.MaxValue)] int skip,
+        [FromQuery, Range(1, 100)] int take)
+    {
+        var (messages, totalCount) = await _messageService.SearchAsync(skip, take, chatId);
+
+        Response.Headers.Add("X-TotalCount", totalCount.ToString());
+
+        return Ok(messages);
+    }
+
+    /// <summary>
+    /// Send new message
+    /// </summary>
+    /// <param name="model">message model</param>
+    /// <response code="200">New message</response>
+    /// <response code="400">Problem details</response>
+    [HttpPost]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    public async Task<ActionResult> SendMessageAsync([FromBody] CreateMessageModel model)
+    {
+        var (result, message) = await _messageService.SendAsync(model.ChatId, model.SenderId, model.Message);
+
+        if (!result.Successed)
+        {
+            return BadRequest(result.Errors);
```

Comment:
We are using rfc7807 (Problem details) as bad request response all over services
See example at: https://github.com/ghosts-network/content/blob/master/GhostNetwork.Content.Api/Helpers/DomainResultExtensions.cs